### ISSUE TITLE
removeVirtualItems - Remove case sensitivity

### ISSUE THREAD

**Issue by johnb432** [acemod/ACE3#8844]
**Is your enhancement related to a problem?**
The ACE Arsenal doesn't return case sensitive naming for weapon attachments (which is another thing that could be remedied, but that's not the point of the issue).
If one wants to remove virtual items from the ACE Arsenal, they must give the case sensitive config name of the items to be removed.

E.g

`[_this, ["optic_DMS"]] call ace_arsenal_fnc_removeVirtualItems` does not work, whereas
`[_this, ["optic_dms"]] call ace_arsenal_fnc_removeVirtualItems` does.

This means the user has to look it up in the configs. Frankly it's not a big deal, but it would be nice if that could be avoided.

**Solution you'd like:**
I would like to see it case insensitive. It would be a quality of life improvement.

**Proposed solution :**

https://github.com/acemod/ACE3/blob/f85ed213c59ba6926c230e0ef8ca1ead292b53af/addons/arsenal/functions/fnc_removeVirtualItems.sqf#L55

```SQF
// Make sure all items are in string form
_items = _items select {_x isEqualType "" && {_x != ""}};

// Find the items that are to be removed using non-case sensitive means
private _item = "";

private _fnc_remove = {
    {
        _item = _x;
        _this deleteAt (_this findIf {_x == _item});
    } forEach _items;

    _this;
};

{
    if (_forEachIndex isEqualTo 0) then {
        _cargo set [_forEachIndex, [(_x select 0) call _fnc_remove, (_x select 1) call _fnc_remove, (_x select 2) call _fnc_remove]];
    } else {
        if (_forEachIndex isEqualTo 1) then {
            _cargo set [_forEachIndex, [(_x select 0) call _fnc_remove, (_x select 1) call _fnc_remove, (_x select 2) call _fnc_remove, (_x select 3) call _fnc_remove]];
        } else {
            _cargo set [_cargo find _x, _x call _fnc_remove];
        };
    };
} forEach _cargo;
```
This is what I have come up with as a working solution. However, it's noticeably slower for large arrays:
Testing with about 700 entries in the items parameter made the game freeze for an estimated time of 1-2s.

I'm hoping others can (help) implement this enhancement.

**Comment by rautamiekka** [comment]
Since I can't test I'll have to speculate, but would it be faster, if less RAM-efficient, to return a copy that's been created by adding only the wanted stuff, instead of deleting from the original array ?

**Comment by commy2** [comment]
Oof, this is terrible code.

Wouldn't it be enough to just lower case everything?

```sqf
_items = _items apply {toLower _x};
```
in L56.

`optic_DMS` is the actual config case. If I were alganthe, I would've used that. Isn't that what is used here? In case you made a mistake, converting everything to config case is:

```sqf
_items = _items apply {configName (_x call CBA_fnc_getItemConfig)};
```

**Comment by commy2** [comment]
Should the case be random (I don't believe that), a much better way to write _fnc_remove is:

```sqf
private _fnc_remove = {_this select {toLower _x in _items}};
```
(which requires the first example I gave to sanitize _items).

Last resort would be inlining the filter via macro:

```
#define REMOVE_ITEMS(from, what) ((from) select {toLower _x in (what)})
```

and then
```sqf
(_x select 0) - _items
```
->
````sqf
REMOVE_ITEMS(_x select 0, _items)
```

**Comment by johnb432** [comment]
> ```sqf
> _items = _items apply {configName (_x call CBA_fnc_getItemConfig)};
> ```
This is exactly what is needed, thanks!

**Comment by LinkIsGrim** [comment]
Addressed in #9040.
